### PR TITLE
Delete activity

### PR DESCRIPTION
**Pull request by anantmittal**

### CODE REVIEW

File: scope_shared/scope/testing/fake_data/fixtures_fake_scheduled_activities.py [hunk 1]
```diff
@@ -169,6 +169,7 @@ def fixture_data_fake_scheduled_activities_factory(
     # Scheduled activities are not defined in such a scenario.
     # Generate new activity schedules that includes at least one element.
     activity_schedules = data_fake_activity_schedules_factory()
+    # NOTE: Check w/ James if this can go away now.
```
Comment (anantmittal) [inline scope_shared/scope/testing/fake_data/fixtures_fake_scheduled_activities.py:172]: @jayfo Left a small note for discussion here.

**Comment by anantmittal** [comment]
@jayfo Ready for review.

File: scope_shared/scope/testing/test_database/test_delete_activity_schedule_maintains_scheduled_activities.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        )
+    )
+    assert get_fake_activity_schedule == None
+
+    # Get scheduled activities
+    assert (
+        scope.database.patient.scheduled_activities.get_scheduled_activities(
+            collection=patient_collection
+        )
+        == None
```
Comment (anantmittal) [inline scope_shared/scope/testing/test_database/test_delete_activity_schedule_maintains_scheduled_activities.py:81]: @jayfo `[]` will replace `None` here.

File: scope_shared/scope/database/patient/activity_schedules.py [hunk 3]
```diff
@@ -148,6 +148,52 @@ def _maintain_pending_scheduled_activities(
...
+
+    if result.inserted_count == 1:
+        existing_scheduled_activities = (
+            scope.database.patient.scheduled_activities.get_scheduled_activities(
+                collection=collection
+            )
+        )
+
+        # Perform only the delete component of maintenance
+        delete_items = _calculate_scheduled_activities_to_delete(
```
Comment (anantmittal) [inline scope_shared/scope/database/patient/activity_schedules.py:178]: @jayfo Not completely sure if the test in `test_delete_activity_schedule...` is covering all the edge cases?

Comment (jayfo) [reply scope_shared/scope/database/patient/activity_schedules.py:178]: I agree, but those same code paths are more explicitly tested in `test_activity_schedule_maintains_scheduled_activities.py`. Since I'm re-using them, this felt fine.
